src/cargo/ops/cargo_generate_lockfile.rs

```diff
@@ -106,4 +107,27 @@ pub fn update_lockfile(manifest_path: &Path,
             None => {}
         }
     }
+
+    fn compare_dependency_graphs<'a>(previous_resolve: &'a Resolve, resolve: &'a Resolve) {
+        let mut dependencies = HashMap::new();
+        for dep in resolve.iter() {
+            dependencies.insert(dep.name(), dep.version());
+        }
+        for dep in previous_resolve.iter() {
+            match dependencies.get(dep.name()) {
+                Some(&version) => {
+                    if version != dep.version() {
+                        println!("    Updating {} v{} -> v{}", dep.name(), dep.version(), version)
```

Comment:
Could this thread through the `Shell` from `UpdateOptions` instead of printing directly to stdout?